front/app/modules/commercial/insights/admin/containers/Insights/Details/Network/index.tsx

```diff
@@ -144,35 +165,144 @@ const Network = ({
       ctx.textAlign = 'center';
       ctx.textBaseline = 'middle';
       ctx.fillStyle = nodeColors[node.color_index % nodeColors.length];
+      const textWidth = ctx.measureText(label).width;
+      node.nodeVerticalOffset = nodeVerticalOffset;
+      node.textWidth = textWidth;
+      node.globalScale = globalScale;
+      node.nodeFontSize = nodeFontSize;
 
       if (globalScale >= visibleKeywordLabelScale) {
         ctx.fillText(label, node.x, nodeVerticalOffset);
+
+        if (node == hoverNode) {
+          // Draw hide icon
+          const hideIcon = new Path2D(
+            'M7.84 6.5l4.89-4.84c.176-.174.274-.412.27-.66 0-.552-.447-1-1-1-.25.003-.488.107-.66.29L6.5 5.13 1.64.27C1.47.1 1.24.003 1 0 .448 0 0 .448 0 1c.01.23.105.45.27.61L5.16 6.5.27 11.34c-.177.173-.274.412-.27.66 0 .552.448 1 1 1 .246-.004.48-.105.65-.28L6.5 7.87l4.81 4.858c.183.184.433.28.69.27.553 0 1-.446 1-.998-.01-.23-.105-.45-.27-.61L7.84 6.5z'
+          );
+          const p = new Path2D();
+          const transform = {
+            a: 0.8 / globalScale,
+            d: 0.8 / globalScale,
+            e: node.x + textWidth / 2 + 5 / globalScale,
+            f: nodeVerticalOffset - 6 / globalScale,
+          };
+          p.addPath(hideIcon, transform);
+          ctx.fillStyle = colors.label;
+          ctx.fill(p);
+
+          // Change tooltip text on icon hover
+          const [offsetX, offsetY] = pointerPosition;
+          const rect = new Path2D();
+          rect.rect(
+            node.x + textWidth / 2 + 5 / globalScale,
+            nodeVerticalOffset - 6 / globalScale,
+            12 / globalScale,
+            12 / globalScale
+          );
+
+          if (ctx.isPointInPath(rect, offsetX, offsetY)) {
+            tooltipRef.textContent = formatMessage(messages.networkHideNode);
+          } else {
+            tooltipRef.textContent = label;
+          }
+        }
       }
     }
   };
 
-  const handleNodeClick = (node: Node) => {
+  const handleNodeClick = (node: Node, event) => {
+    let removedNode = false;
```

Comment:
What is this doing?

Reply:
removedNode is a flag to detect wheter the click was in the hide icon, so later in the function it could be removed from the selected keywords. it actually could be named something like `hiddenNode` or `isHidden`